### PR TITLE
Orient schema migrations

### PR DESCRIPTION
**Pull request by yusefnapora**
This adds some helpers for creating vertex and edge classes, and uses [this fork](https://github.com/mediachain/orientdb-migrations/tree/mediachain) of the [orientdb-migrations](https://github.com/springnz/orientdb-migrations) project to apply migrations.

The fork is mostly there because I couldn't find a published jar anywhere, and I needed to fix up a couple of things before adding it as a git dependency to sbt.  In theory we could publish it ourselves somewhere, but I don't know enough about maven and friends to make that happen.

Anyway, the `io.mediachain.util.orient.LSpaceMigrations` class has a migration that creates the vertex and edge classes we're using in `master` at the moment.  It uses some helpers defined in `io.mediachain.util.orient.OrientSchema`, which has "builders" for vertex and edge classes, as well as `StringProperty`, `BoolProperty`, etc.

There's a `MigrationHelper.newInMemoryGraph` function that will create a new in-memory db and migrate it before returning the graph, which I've updated the specs to use.  By default it returns a transactional graph, which seems to work without blowing anything up.

The migrations are recorded in the database, so re-applying them should only run the ones you haven't applied yet.  There's no rollback feature; the migrations are basically just side-effecting functions that accept an `ODatabaseDocumentTx`, so if you need to undo something, you have to do it in the next migration.

I added a fn called `MigrationHelper.applyToDBConfigAtPath` which takes the path to a config file (in typesafe's config format).  I haven't tried it out yet, but it should come in handy when we set up the persistent db.  It just calls through to the code in orientdb-migrations; the config loading code is [here](https://github.com/springnz/orientdb-migrations/blob/master/src/main/scala/springnz/orientdb/pool/ODBConnectionPool.scala#L19)

### CODE REVIEW

File: l_space/src/main/scala/io/mediachain/util/orient/LSpaceMigrations.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      VertexClass("Person",
+        StringProperty("name").mandatory(true).readOnly(true)
+      ),
+
+      VertexClass("RawMetadataBlob",
+        StringProperty("blob").mandatory(true).readOnly(true)
+      ),
+
+      EdgeClass(DescribedBy),
```
Comment (parkan) [inline l_space/src/main/scala/io/mediachain/util/orient/LSpaceMigrations.scala]: Having slightly nightmarish thoughts about what happens in the distributed future if these diverge between nodes

File: l_space/src/main/scala/io/mediachain/util/orient/MigrationHelper.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      override def dbConfig: Try[ODBConnectConfig] =
+        for {
+          url <- getEnv("ORIENTDB_URL")
+          user <- getEnv("ORIENTDB_USER")
+          pass <- getEnv("ORIENTDB_PASS")
+        } yield ODBConnectConfig(url, user, pass)
+    }
+
+
+  def getEnv(key: String): Try[String] =
```
Comment (parkan) [inline l_space/src/main/scala/io/mediachain/util/orient/MigrationHelper.scala]: should stick this in utils

File: l_space/src/main/scala/io/mediachain/util/orient/OrientSchema.scala [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      isReadOnly = r
+      this
+    }
+
+    def unique(u: Boolean): this.type = {
+      createUniqueIndex = u
+      this
+    }
+
+    def addTo(cls: OClass): OProperty = {
```
Comment (parkan) [inline l_space/src/main/scala/io/mediachain/util/orient/OrientSchema.scala]: I myself like "klass" :stuck_out_tongue:

File: l_space/src/test/scala/io/mediachain/LSpaceSpec.scala [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package io.mediachain
 
+import _root_.io.mediachain.util.orient.MigrationHelper
```
Comment (parkan) [inline l_space/src/test/scala/io/mediachain/LSpaceSpec.scala]: why does this need a qualified absolute import?

Comment (yusefnapora) [inline l_space/src/test/scala/io/mediachain/LSpaceSpec.scala]: oh weird.  I'm guessing that was some intellij craziness that slipped past me

File: project/Build.scala [hunk 5]
```diff
@@ -48,6 +49,14 @@ object LSpaceBuild extends Build{
...
+  val orientdb_migrations_commit = "5f345cefda34f5671e6bb9e6c30312299d11f371"
+  lazy val orientdb_migrations = ProjectRef(
+    uri("git://github.com/mediachain/orientdb-migrations.git#" +
+      orientdb_migrations_commit),
+    "orientdb-migrations-root"
+  )
+
+
   lazy val root = (project in file("."))
     .aggregate(core, l_space,
```
Comment (parkan) [inline project/Build.scala]: should probably aggregate this so it runs tests from the root, unless you don't want that?

Comment (yusefnapora) [inline project/Build.scala]: you mean the tests for the orientdb-migrations project?

Comment (parkan) [inline project/Build.scala]: yeah

Comment (yusefnapora) [inline project/Build.scala]: ah yeah, I bailed out of that when circle got confused about something to do with scalatest and JUnit.... I can try to sort that out again

Comment (parkan) [inline project/Build.scala]: seems to work for me so I checked it in

Comment (yusefnapora) [inline project/Build.scala]: awesome, thanks

**Pull request merged**